src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -415,6 +424,10 @@ class StubGenerator: public StubCodeGenerator {
 
     __ ld(x9, x9_save);
 
+    // restore fcsr
+    __ ld(t1, fcsr_save);
+    __ csrw(CSR_FCSR, t1);
```

Comment:
It would be better to avoid the `csrw` in case the CSR is already set to `RoundingMode::rne` on some hardware. You can have avoid it with a simple check on the current value of FCSR.

Reply:
Interesting. So is it safe to claim that `csrr` is faster than `csrw`? I didn't measure the difference.

Reply:
It's fair to claim that `csrw` has side effects which may be more detrimental to performance, side effects that `csrr` that doesn't have (and can't as it's only reading).

Reply:
Another thing that is worth considering is that the `fflags` (exception flags) contained in `fcsr` is very likely to change after the Java call which does floating point calculations.

Reply:
Yes, that make sense to me. I think `fflags` is safe to ignore here.

Reply:
Thanks for discussion, all updated.